### PR TITLE
Support custom steps entrypoint

### PR DESCRIPTION
**Pull request by qwerty287**
Closes https://github.com/woodpecker-ci/woodpecker/issues/278

### CODE REVIEW

**Review by 6543 — CHANGES_REQUESTED** [review]
It was deliberatly removed because of security concerns.

If added there need some other checks made too

File: pipeline/backend/kubernetes/pod_test.go [hunk 1]
```diff
@@ -236,7 +237,7 @@ func TestFullPod(t *testing.T) {
 
 	pod, err := mkPod("woodpecker", "wp-01he8bebctabr3kgk0qj36d2me-0", "meltwater/drone-cache", "/woodpecker/src", "linux/amd64", "wp-svc-acc",
 		true, true,
-		[]string{"go get", "go test"}, []string{"woodpecker-cache:/woodpecker/src/cache"}, []string{"cloudflare:1.1.1.1"},
+		[]string{"go get", "go test"}, []string{"woodpecker-cache:/woodpecker/src/cache"}, []string{"cloudflare:1.1.1.1"}, nil,
```
Comment (zc-devs) [inline pipeline/backend/kubernetes/pod_test.go:240]: In this test an entrypoint should be tested. Set some example value.

**Comment by 6543** [comment]
plugins should not count as plugins anymore if the entrypoint ha schanged

**Comment by 6543** [comment]
and the pull title is missleding, when i was reading it, i thought "oh woodpecker server gets an custom endpoint?!?" - well if you read carefull you might get it.

why not put "steps" upfront?

**Comment by 6543** [comment]
:point_up: yes that title is nide :heart:  - beside the pull needs conflict resolve :sweat:

File: docs/docs/20-usage/20-workflow-syntax.md [hunk 2]
```diff
@@ -154,6 +154,10 @@ docker run --entrypoint=build.sh golang
 
 > Please note that only build steps can define commands. You cannot use commands with plugins or services.
 
+### `entrypoint`
+
+Allows you to specify the entrypoint for Docker and Kubernetes. Note that this must include the full command as list, including arguments (e.g. `["/bin/sh", "-c"]`).
```
Comment (anbraten) [inline docs/docs/20-usage/20-workflow-syntax.md:159]:
```suggestion
Allows you to specify the entrypoint for containers. Note that this must be a list of the command and its arguments (e.g. `["/bin/sh", "-c"]`).
```

File: pipeline/backend/kubernetes/pod.go [hunk 3]
```diff
@@ -124,6 +124,9 @@ func podContainer(name, image, workDir, goos string, pull, privileged bool, comm
 
 	if len(commands) != 0 {
 		scriptEnv, command, args := common.GenerateContainerConf(commands, goos)
+		if len(entrypoint) > 0 {
+			command = entrypoint
```
Comment (anbraten) [inline pipeline/backend/kubernetes/pod.go:128]: What happens to the arguments?

File: pipeline/backend/local/local.go [hunk 4]
```diff
@@ -146,6 +146,7 @@ func (e *local) StartStep(ctx context.Context, step *types.Step, taskUUID string
 // execCommands use step.Image as shell and run the commands in it
 func (e *local) execCommands(ctx context.Context, step *types.Step, state *workflowState, env []string) error {
 	// Prepare commands
+	// TODO use `entrypoint` from yaml
```
Comment (anbraten) [inline pipeline/backend/local/local.go:149]: Was at the beginning a bit confused which yaml you refer to 😅 

```suggestion
	// TODO support `entrypoint` from pipeline config
```

File: pipeline/frontend/yaml/types/container.go [hunk 5]
```diff
@@ -121,7 +122,7 @@ func (c *ContainerList) UnmarshalYAML(value *yaml.Node) error {
 }
 
 func (c *Container) IsPlugin() bool {
-	return len(c.Commands) == 0
+	return len(c.Commands) == 0 && len(c.Entrypoint) == 0
```
Comment (6543) [inline pipeline/frontend/yaml/types/container.go:125]: can we have a test for this?

**Review by 6543 — APPROVED** [review]

**Pull request merged**